Batteries/Data/List/Lemmas.lean

```diff
@@ -482,6 +491,34 @@ theorem Sublist.erase_diff_erase_sublist {a : α} :
 
 end Diff
 
+/-! ### prefix, suffix, infix -/
+
+theorem singleton_prefix_cons (a) : [a] <+: a :: l :=
+  (prefix_cons_inj a).mpr nil_prefix
```

Comment:
If suitable `alias`es were added, this could be written `.cons a .nil`, which is then shorter than the lemma name.

Reply:
Can you show me how to add those aliases? I'm not sure how to do it.

Reply:
I'd like to hear what other maintainers think about adding these aliases.

Reply:
These types of aliases never really caught on outside Mathlib as far as I can tell. They're convenient for sure but I'm not sure that kind of convenience is a good fit for Batteries. I would rather see more automation than these types of shortcuts as far as Batteries is concerned.

Reply:
Here's an argument for why we should have them; you could imagine an alternative world where `IsPrefix` is defined as
```lean
inductive IsPrefix where
| nil {l} : IsPrefix [] l
| cons (a) {l1 l2} : IsPrefix l1 l2 -> IsPrefix (a :: l1) (a :: l2)
```

The API should be agnostic to the actual implementation chosen, and so we should provide `nil` and `cons`  anyway.

Reply:
@eric-wieser The relation `List.IsPrefix` is defined in Lean's core library. Where should we put the aliases? Lean's core or Batteries?

Reply:
That's a fair point @eric-wieser ! However, in that alternate world, I might think that since `IsPrefix` is a `Prop`, its constructors should follow proper theorem naming conventions...

Anyway, I'm not vehemently opposed to these aliases, I just don't think they're necessary.

Reply:
Should be `nil_isPrefix` and `cons_isPrefix_cons`, no?